### PR TITLE
Check error from snapshots() function

### PR DESCRIPTION
**Pull request by Vampouille**
Also checks that there is at least one snapshots

### CODE REVIEW

File: engines/restic.go [hunk 1]
```diff
@@ -211,6 +212,13 @@ func (r *ResticEngine) forget() (err error) {
 	v := r.Volume
 
 	snapshots, err := r.snapshots()
+	if err != nil {
+		return
+	}
+	if len(snapshots) == 0 {
+		err = errors.New("No snapshots found but bucket should contains at least current backup")
```
Comment (mcanevet) [inline engines/restic.go]: please use `log.Errorf()` or `log.Fatalf()`

Comment (mcanevet) [inline engines/restic.go]: please use log.Errorf() or log.Fatalf()

**Comment by mcanevet** [comment]
Somehow tests fails...

File: engines/restic.go (hunk 1 above)
Comment (Vampouille) [reply engines/restic.go]: log.Errorf() just generate a message on stderr, here we want to create a error var.

**Comment by Vampouille** [comment]
Travis error : 
```
DEBU[0008] Launching with '-r file:///root/.cache/duplicity/.test_backup/0bcdaa5b2b3a/test snapshots --no-cache'... 
DEBU[0009]  Fatal: parsing repository location failed: invalid backend
 If the repo is in a local directory, you need to add a `local:` prefix
```

**Pull request merged**